python/src/iceberg/types.py

```diff
@@ -15,157 +15,1215 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from typing import Optional
+import decimal
+import math
+import struct
+from base64 import b64encode
+from datetime import date, datetime, time
+from decimal import Decimal as PythonDecimal
+from typing import Any, Dict
+from typing import List as PythonList
+from typing import Optional, Tuple, Type, Union
+from uuid import UUID as PythonUUID
 
+import mmh3
+from numpy import float32, float64, isinf, isnan
 
-class Type(object):
-    def __init__(self, type_string: str, repr_string: str, is_primitive=False):
-        self._type_string = type_string
-        self._repr_string = repr_string
-        self._is_primitive = is_primitive
 
-    def __repr__(self):
-        return self._repr_string
+# intended for use inside this module only
+class IcebergMetaType(type):
+    """
+    MetaClass to:
```

Comment:
nit: the current convention is to start the comment without new line, e.g. `"""MetaClass to:`